rotkehlchen/tests/api/test_bitcoin.py

```diff
@@ -320,22 +327,23 @@ def test_add_delete_xpub_multiple_chains(rotkehlchen_api_server):
     # Check that addresses that have balances on both bitcoin and bitcoin cash are stored properly
     with rotki.data.db.conn.read_ctx() as cursor:
         mutual_in_chains_aggregator = 0
-        for btc_addr in rotki.chains_aggregator.accounts.btc:
+        btc_addesses: tuple[BTCAddress, ...] = rotki.chains_aggregator.accounts.btc
```

Comment:
1. you have a typo. `btc_addesses`
2. This should not be needed. This is the type of accounts.btc. Something else is off.

Reply:
I was able to make this simpler by just going `btc_addr: BTCAddress`, but I haven't been able to find any more elegant solution.

Reply:
I changed this by:

```python
    reveal_type(rotki.chains_aggregator.accounts.btc)
    assert len(rotki.chains_aggregator.accounts.btc) == 0
    reveal_type(rotki.chains_aggregator.accounts.btc)
```

I was looking at the types as seen by mypy. You can do reveal_type() and run mypy at any point. Will tell you what type mypy thinks it has at that line. I surrounded this line: https://github.com/rotki/rotki/blob/57f687e1192c8d3c6e66616e0e81e1093d5c0736/rotkehlchen/tests/api/test_bitcoin.py#L292

Result is this:

rotkehlchen/tests/api/test_bitcoin.py:299: note: Revealed type is "builtins.tuple[rotkehlchen.types.BTCAddress, ...]"
rotkehlchen/tests/api/test_bitcoin.py:301: note: Revealed type is "tuple[()]"


So ... before the check the type inference is correct. After the length == 0 type inference is that now it's an empty tuple. Which is also correct. But it does not understand the type has changed later down where you needed it.

But this uncovers the problem. I think the original types are off. They should not be tuples. Tuples are immutable